pkg/core/alias.go

```diff
@@ -22,24 +22,47 @@ var (
 	SetLogger             = kube_log.SetLogger
 	Now                   = time.Now
 
-	SetupSignalHandler = func() (context.Context, context.Context) {
+	SetupSignalHandler = func() (context.Context, context.Context, <-chan struct{}) {
 		gracefulCtx, gracefulCancel := context.WithCancel(context.Background())
 		ctx, cancel := context.WithCancel(context.Background())
 		c := make(chan os.Signal, 3)
-		signal.Notify(c, syscall.SIGINT, syscall.SIGTERM)
+		signal.Notify(c, syscall.SIGINT, syscall.SIGTERM, syscall.SIGUSR2)
+		usr2Notify := make(chan struct{}, 1)
```

Comment:
I'd drop `, 1`. Why buffer if we only write in a non-blocking way

Reply:
Agreed with this understanding